pkg/cmd/server/start/start_node.go

```diff
@@ -224,6 +235,17 @@ func (o NodeOptions) RunNode() error {
 	if addr := o.NodeArgs.ListenArg.ListenAddr; addr.Provided {
 		nodeConfig.ServingInfo.BindAddress = addr.HostPort(o.NodeArgs.ListenArg.ListenAddr.DefaultPort)
 	}
+	// do a local resolution of node config DNS IP, supports bootstrapping cases
+	if nodeConfig.DNSIP == "0.0.0.0" {
```

Comment:
So this is not a listen address, it is a connection address?   Using `0.0.0.0` to do defaulting feels wrong.  `***autodetect***` is an illegal value for a hostname or IP that will never be confused as something else.

Reply:
0.0.0.0 was never allowed before.  We changed it in 3.7 to mean "the local address".  It has to be an IP because of DNS.  We can't change the value that means autodetect because 3.7 clusters are using 0.0.0.0